pilot/cmd/pilot-agent/status/server.go

```diff
@@ -169,6 +172,15 @@ func NewServer(config Config) (*Server, error) {
 		if prober.HTTPGet.Port.Type != intstr.Int {
 			return nil, fmt.Errorf("invalid prober config for %v, the port must be int type", path)
 		}
+		// Construct a http client and cache it in order to reuse the connection.
+		s.appProbeClient[path] = &http.Client{
```

Comment:
Why not use the same client for all paths? The client seem to be equivalent.

Reply:
As you can see below, the timeout maybe different

Reply:
I see. thanks!